csm/core/controllers/information.py

```diff
@@ -77,3 +76,90 @@ async def post(self):
             Log.error(f"Error in checking compatability: {e}")
             raise CsmInternalError(f"{e}")
         return CsmResponse(response)
+
+@CsmAuth.public
+@CsmView._app_routes.view("/api/v2/system/topology")
+class Topology(CsmView):
+    """
+    GET: Complete deployment topology
+    """
+    def __init__(self, request):
+        super().__init__(request)
+        self._service = self.request.app[const.INFORMATION_SERVICE]
+
+    async def get(self):
+        """GET REST implementation for complete topology."""
+        Log.info(f"Processing request: {self.request.method} {self.request.path}")
+        # Call Information Service
+        try:
+            response = await self._service.get_topology()
+        except CsmInternalError as e:
+            raise e
+        except Exception as e:
+            Log.error(f'Unable to fetch topology information: {e}')
+            raise CsmInternalError("Unable to fetch topology information.")
+        Log.info(
+            f"Processed request: {self.request.method} {self.request.path}")
+        return CsmResponse(response)
+
+@CsmAuth.public
+@CsmView._app_routes.view("/api/v2/system/topology/{resource}")
+class ResourceTopology(CsmView):
+    """
+    GET: Get topology of resource specified.
+    """
+    def __init__(self, request):
+        super().__init__(request)
+        self._service = self.request.app[const.INFORMATION_SERVICE]
+
+    async def get(self):
+        """GET REST implementation to query information about resource from deployment topology."""
+        Log.info(f"Processing request: {self.request.method} {self.request.path}")
+        resource = self.request.match_info[const.ARG_RESOURCE]
+        # Check for valid Resource
+        if resource not in const.TOPOLOGY_RESOURCES:
+            raise CsmNotFoundError(f"Invalid resource: {resource}")
+        # Call Information Service
+        Log.info(f"Fetching deployment topology for resource:{resource}.")
+        try:
+            response = await self._service.get_resource(resource)
+        except CsmInternalError as e:
+            raise e
+        except Exception as e:
+            Log.error(f'Unable to fetch topology information: {e}')
+            raise CsmInternalError("Unable to fetch topology information.")
+        Log.info(
+            f"Processed request: {self.request.method} {self.request.path}")
+        return CsmResponse(response)
+
+@CsmAuth.public
+@CsmView._app_routes.view("/api/v2/system/topology/{resource}/{resource_id}")
```

Comment:
Resource ID will not be present for certificate. Have you handled it?

Reply:
yes